### PR TITLE
fix(ConfigFile): correct resolve *-Info.plist file path

### LINKED ISSUE #88: ConfigFIle.js for iOS Info.plist when the CocoaPods version is 1.6.* or later
# Bug Report

Image the `config.xml` or `plugin.xml` includes `config-file` tag like
```
      <config-file target="*-Info.plist" parent="SampleTest">
        <array>
          <string>hello</string>
        </array>
      </config-file> 
```

Then the `cordova-common` searches `[Application Name]-Info.plist` file under the `platform/ios` in order to update `[Application Name]-Info.plist` file in `cordova prepare` process.

This logic can be found in `src/ConfigChanges/ConfigFile.js` in `cordova-common` as

```
       // [CB-5989] multiple Info.plist files may exist. default to $PROJECT_NAME-Info.plist
        if (matches.length > 1 && file.includes('-Info.plist')) {
            var plistName = getIOSProjectname(project_dir) + '-Info.plist';
            for (var i = 0; i < matches.length; i++) {
                if (matches[i].includes(plistName)) {
                    filepath = matches[i];
                    break;
                }
            }
        }
```

This works well if the cocoapods version is 1.5.* or earlier.

However if the cocoapods version is 1.6.* or later, the project includes plugins which include cocoapods libraries with subdirectory path like `<pod name="Firebase/Core" spec="6.3.0"/>` , the file
```
  platforms/ios/Pods/Target Support Files/Pods-[Application Name]/Pods-[Application Name]-Info.plist
```
is created and this may lead wrong behavior in `config-file` tag with target `*-Info.plist` in `cordova prepare ios`.

This created file satisfies the above logic (i.e. finding `*-Info.plist` file) condition.

(Rematk that If the cocoapods version is 1.5.* or earlier, that file name is different. Actually the created file in this case is 

```
  platforms/ios/Pods/Target Support Files/Pods-[Application Name]/Info.plist
```
.  Therefore no problem happens)

## Problem

If the build environment uses CocoaPods 1.6.* or later and the project uses CocoaPods Library with subdirectory path structure like `<pod name="Firebase/Core" spec="6.3.0"/>`, 
There are two files which satisfies the above logic condition (i.e. `*-Info.plist` file finding condition) in config-file.  That two files' path are

```
platforms/ios/[Application Name]-Info.plist
platforms/ios/Pods/Target Support Files/Pods-[Application Name]/Info.plist
```

If the first letter of [Application Name] is `Q-Z` (i.e. the letter after `P` of `Pods`) , then the above logic detect the wrong file

```
 platforms/ios/Pods/Target Support Files/Pods-[Application Name]/Info.plist
```

instead of the correct file

```
platforms/ios/[Application Name]-Info.plist
```

as its `target file`.

### What is expected to happen?

Plugin's `config-file` with the target `*-Info.plist` must be applied to the correct one
```
platforms/ios/[Application Name]-Info.plist
```

### What does actually happen?

Plugin's `config-file` with the target `*-Info.plist` is applied to
```
platforms/ios/Pods/Target Support Files/Pods-[Application Name]/Info.plist
```

If all followings are satisfied,

i) cocoapods version is 1.6.* or later,
ii) some cocoapods libraries use subdirectory path structure. (like `cordova-plugin-firebasex` plugin. This plugin is different from `cordova-plugin-firebase`. )
iii) `config-file` tag in `config.xml` or `plugin.xml` changes `*-Info.plist` file.

### Command or Code

How to reproduce this iusse,

1. create cordova project
```
$ cordova create sampleProject
```

2. replace package.json
```
{
  "name": "helloworld",
  "displayName": "HelloCordova",
  "version": "1.0.0",
  "description": "A sample Apache Cordova application that responds to the deviceready event.",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [
    "ecosystem:cordova"
  ],
  "author": "Apache Cordova Team",
  "license": "Apache-2.0",
  "dependencies": {
    "cordova-ios": "^5.0.1",
    "cordova-plugin-androidx": "^1.0.2",
    "cordova-plugin-androidx-adapter": "^1.1.0",
    "cordova-plugin-firebasex": "^6.0.4"
  },
  "devDependencies": {
    "cordova-plugin-whitelist": "^1.3.4"
  },
  "cordova": {
    "plugins": {
      "cordova-plugin-whitelist": {},
      "cordova-plugin-firebasex": {}
    },
    "platforms": [
      "ios"
    ]
  }
}
```

3. update config.xml

```
<widget ...>
    <name>Z-HelloCordova</name>
   ...
    <platform name="ios">
      <config-file target="*-Info.plist" parent="SampleTest">
        <array>
          <string>hello</string>
        </array>
      </config-file> 
      <allow-intent href="itms:*" />
      <allow-intent href="itms-apps:*" />
    </platform>     
```

4. do cordova prepare

```
$ cordova prepare ios
```

Then thare is no `SampleTest` in `platforms/ios/Z-HelloCordova/Z-HelloCordova-Info.plist`

However if we use the application name `A-HelloCordova`
```
    <name>A-HelloCordova</name>
```
in step2
, there is `SampleTest` in `platforms/ios/A-HelloCordova/A-HelloCordova-Info.plist` as

```
	<key>SampleTest</key>
	<array>
		<string>hello</string>
	</array>
```

### Environment, Platform, Device

Mac
ruby 2.3.7p456
CocoaPods 1.7.5
xcodebuild 10.3 (Build Version 10G8)
(Remark that this issue happens in `cordova prepare ios`)

### Version information
cordova-cli 9.0.0 (cordova-lib@9.0.1)
cordova-ios (in project node_modules) 5.0.1
cordova-common (in project node_modules) 3.2.0

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

### PR DESCRIPTION
**Pull request by Armanio**
fixes #88

### CODE REVIEW

**Comment by Armanio** [comment]
@raphinesse thanx for review. What about updated PR? 🤔

**Comment by Armanio** [comment]
@raphinesse its reasonably.  I tried to write test, but im not sure that he is correct. If you can check test and PR it would be great.
And we need check it on old cordova version. Sorry, but I not tested, my bad.

**Comment by brodybits** [comment]
What is the status of this?

**Comment by raphinesse** [comment]
@Armanio Is the current state ready for review or are you still working on this PR?

@brodybits This still needs a review. I'm slow to respond currently as I can put only little time aside for Cordova right now.

**Comment by brodybits** [comment]
Thanks @raphinesse. It looks to me like @Armanio did already push some updates in response to your latest feedback. I am also wondering if this may be related to some of the other issues I discussed in <https://github.com/apache/cordova-ios/issues/793>. Any pointers would be much appreciated since I am still not a super expert with the plist functionality. I do completely understand about the situation with your response time.

**Comment by Armanio** [comment]
@raphinesse yes, its ready to review. I hope all its ok. :relieved:

**Comment by brodybits** [comment]
From a quick looks the changes seem to be OK to me. Unfortunately I still do not understand this part of Cordova all that well and other maintainers are pretty busy with the new major release. As I said before, I did notice some other issues that may be related as discussed in <https://github.com/apache/cordova-ios/issues/793>, would appreciate any pointers to understand if there may be any overlap or not. Thanks!

File: src/ConfigChanges/ConfigFile.js [hunk 1]
```diff
@@ -172,8 +172,9 @@ function resolveConfigFilePath (project_dir, platform, file) {
 
         // [CB-5989] multiple Info.plist files may exist. default to $PROJECT_NAME-Info.plist
         if (matches.length > 1 && file.includes('-Info.plist')) {
-            const plistName = `${getIOSProjectname(project_dir)}-Info.plist`;
-            filepath = matches.find(m => m.includes(plistName)) || filepath;
+            const projName = getIOSProjectname(project_dir);
+            const plistName = projName + '-Info.plist';
```
Comment (erisu) [inline src/ConfigChanges/ConfigFile.js]:
```suggestion
            const plistName = `${projName}-Info.plist`;
```

File: spec/ConfigChanges/ConfigFile.spec.js [hunk 2]
```diff
@@ -97,6 +97,24 @@ describe('ConfigFile tests', function () {
...
+                spyOn(glob, 'sync').and.callFake(arg =>
+                    arg === '*.xcodeproj' ? [projName] : [
+                        path.join('project_dir', 'Pods', 'Target Support Files', 'Pods-' + projName, 'Pods-' + projName + '-Info.plist'),
+                        path.join('project_dir', 'Pods', 'Target Support Files', 'Pods-' + projName, projName + 'Info.plist'),
+                        path.join('project_dir', projName + '-Info.plist'),
+                        path.join('project_dir', projName, projName + '-Info.plist')
+                    ]
+                );
+
+                var plistPath = path.join('project_dir', projName, projName + '-Info.plist');
```
Comment (erisu) [inline spec/ConfigChanges/ConfigFile.spec.js]:
```suggestion
                const plistPath = path.join('project_dir', projName, `${projName}-Info.plist`);
```

File: spec/ConfigChanges/ConfigFile.spec.js [hunk 3]
```diff
@@ -97,6 +97,24 @@ describe('ConfigFile tests', function () {
...
+                spyOn(configFile, 'getIOSProjectname').and.returnValue('AAA');
+                var projName = configFile.getIOSProjectname();
+
+                var glob = require('glob');
+                spyOn(glob, 'sync').and.callFake(arg =>
+                    arg === '*.xcodeproj' ? [projName] : [
+                        path.join('project_dir', 'Pods', 'Target Support Files', 'Pods-' + projName, 'Pods-' + projName + '-Info.plist'),
+                        path.join('project_dir', 'Pods', 'Target Support Files', 'Pods-' + projName, projName + 'Info.plist'),
+                        path.join('project_dir', projName + '-Info.plist'),
+                        path.join('project_dir', projName, projName + '-Info.plist')
```
Comment (erisu) [inline spec/ConfigChanges/ConfigFile.spec.js]:
```suggestion
                        path.join('project_dir', 'Pods', 'Target Support Files', `Pods-${projName}`, `Pods-${projName}-Info.plist`),
                        path.join('project_dir', 'Pods', 'Target Support Files', `Pods-${projName}`, `${projName}Info.plist`),
                        path.join('project_dir', `${projName}-Info.plist`),
                        path.join('project_dir', projName, `${projName}-Info.plist`)
```

Also, is it expected that the second item in the array to not have a `-` in the `Info.plist`? All other items had a `-` separator.

File: spec/ConfigChanges/ConfigFile.spec.js [hunk 4]
```diff
@@ -97,6 +97,24 @@ describe('ConfigFile tests', function () {
                 var configPath = path.join('project_dir', 'app', 'src', 'main', file, 'xml');
                 expect(configFile.resolveConfigFilePath('project_dir', 'android', file)).toBe(configPath);
             });
+
+            it('should return *-Info.plist file', function () {
+                spyOn(configFile, 'getIOSProjectname').and.returnValue('AAA');
+                var projName = configFile.getIOSProjectname();
```
Comment (erisu) [inline spec/ConfigChanges/ConfigFile.spec.js]:
```suggestion
                const projName = configFile.getIOSProjectname();
```

File: spec/ConfigChanges/ConfigFile.spec.js [hunk 5]
```diff
@@ -97,6 +97,24 @@ describe('ConfigFile tests', function () {
...
+
+                var glob = require('glob');
```
Comment (erisu) [inline spec/ConfigChanges/ConfigFile.spec.js]:
```suggestion
                const glob = require('glob');
```

**Pull request merged**